lib/Onyx.js

```diff
@@ -195,7 +195,10 @@ function keysChanged(collectionKey, collection) {
                     newData = lodashMerge(newData, prevState[subscriber.statePropertyName]);
                 }
                 return {
-                    [subscriber.statePropertyName]: newData,
+                    // If `newData` happens to not exist, then set the state back to the previous value because
+                    // if the state is set to an undefined value, then the item reverts to whatever is in
+                    // defaultProps
+                    [subscriber.statePropertyName]: typeof newData !== 'undefined' ? newData : prevState[subscriber.statePropertyName],
```

Comment:
Should we use `_.isUndefined()` ?

Reply:
Ah, yeah... that's a good call. I guess we do have underscore here.